File: engine/model/src/main/scala/net/server/channel/packet/bbs/NewBBSThreadPacket.scala
package net.server.channel.packet.bbs

class NewBBSThreadPacket(private var _mode: Byte, private var _isNotice: Boolean, private var _title: String, private var _text: String, private var _icon: Int) extends BaseBBSOperationPacket(_mode = _mode) {
  def isNotice: Boolean = _isNotice

  def title: String = _title

  def text: String = _text

  def icon: Int = _icon
}

File: engine/engine-model/src/main/java/net/server/channel/packet/bbs/NewBBSThreadPacket.java
package net.server.channel.packet.bbs;

public class NewBBSThreadPacket extends BaseBBSOperationPacket {
   private final Boolean isNotice;

   private final String title;

   private final String text;

   private final Integer icon;

   public NewBBSThreadPacket(Byte mode, Boolean isNotice, String title, String text, Integer icon) {
      super(mode);
      this.isNotice = isNotice;
      this.title = title;
      this.text = text;
      this.icon = icon;
   }

   public Boolean isNotice() {
      return isNotice;
   }

   public String title() {
      return title;
   }

   public String text() {
      return text;
   }

   public Integer icon() {
      return icon;
   }
}
